pandas/core/reshape/merge.py

```diff
@@ -561,6 +565,9 @@ def __init__(self, left, right, how='inner', on=None,
         # to avoid incompat dtypes
         self._maybe_coerce_merge_keys()
 
+        if self.validate is not None:
```

Comment:
add a comment here on what is happening, pass in ``validate`` and don't make it an instance variable. This will just raise if there is an error.